src/main.rs

```diff
@@ -109,6 +197,29 @@ fn main() -> Result<()> {
     let mut buffer = Vec::new();
     input.read_to_end(&mut buffer)?;
 
+    let scaling_factor = match analyze_result {
+        Ok(rate) => {
+            let input: Input = serde_json::from_slice(&buffer)?;
+            let num_cart_lines = input.cart.lines.len();
```

Comment:
We should use the schema to determine which fields to scale on, rather than hardcoding it for `cart.lines`.

Reply:
We can use the `extra_info` added to latest unreleased `bluejay-validator` crate ([here](https://github.com/Shopify/bluejay/blob/895b3783e98bce4e6d9756133fb23351b635de38/bluejay-validator/src/executable/operation/orchestrator.rs#L342)) to pass the JSON input into the analyzer and use that while visiting the query to compute the right limits based on the input.

Reply:
And of course once we're ready to ship, I'll release a new version of `bluejay-validator` so that we don't have to point to a specific git commit and can go through `crates.io`.